### PR TITLE
[Sketcher] Tool parameter : various fixes

### PR DESCRIPTION
**Pull request by PaddleStroke**
@abdullahtahiriyo this PR fixes a first batch of comments that has been made.
It does not implement the preference to disable onviewparameters as you are doing it already.

Fixes issues by @0penBrain 

> * Most severe one is probably the Tab management. Eg. when drawing a rectangle, X coord is the selected one. If I press Tab, I go to Y coord. Fine. But here if I press Tab again, then coord selection is lost forever. I would expect Tab only to circle between currently available coords/dims. Also I'd expect Shift+Tab to work (circling in the other direction).
> * While keeping the coord/dim boxes on the viewport is well handled on top and left, it isn't on right and bottom where they disappear out of view.
> 
> * The mode named "Diagonal corners" isn't. It is "1 corner + length & width". A true "diagonal corners" mode is missing.
> * Same goes for "Center and corner" which actually is "Center + length & width".

Fixes issue from chrisb:

> In certasin positions the dynamically moving input fields are lying over each other. In that case the one having the focus should be on top.

Fixes the issue of spinbox getting in the way raised on the german topic.

Fixes https://github.com/FreeCAD/FreeCAD/issues/11289
Fixes https://github.com/FreeCAD/FreeCAD/issues/11288
Fixes https://github.com/FreeCAD/FreeCAD/issues/11287
Fixes https://github.com/FreeCAD/FreeCAD/issues/11285
Fixes https://github.com/FreeCAD/FreeCAD/issues/11284

### CODE REVIEW

**Comment by PaddleStroke** [comment]
@abdullahtahiriyo that's it I fixed all the issues raised that I could fix.

The things that needs to be fixed still : 
- Preferences to disable OnViewParameter / widget.
- Cursor text (as this is linked to preferences). Also we did not decide what we want to do when enabled.
- @0penBrain mentioned that 3p rectangle is in fact 3p parallelepiped. But I don't think there's anything we can do to improve this. OpenBrain do you have a suggestion or was it just a comment and we leave it like this?

File: src/Mod/Sketcher/Gui/DrawSketchDefaultWidgetController.h [hunk 1]
```diff
@@ -253,25 +253,7 @@ class DrawSketchDefaultWidgetController: public DrawSketchController<HandlerT,
...
-
-            if (currentindex != methodint) {
-                // avoid triggering of method change
-                boost::signals2::shared_connection_block combobox_block(
-                    connectionComboboxSelectionChanged);
-                toolWidget->setComboboxIndex(WCombobox::FirstCombo, methodint);
-            }
-        }
-    }
+    {}
```
Comment (abdullahtahiriyo) [inline src/Mod/Sketcher/Gui/DrawSketchDefaultWidgetController.h:256]: I am curious. 

This you moved to the reset function below. However, I cannot come to a reason why is this change necessary.

 In other words, do we need this code on every reset, even when the construction mode did not change? Is it not only necessary when changing construction mode?

Comment (PaddleStroke) [reply src/Mod/Sketcher/Gui/DrawSketchDefaultWidgetController.h:256]: This is commit  [da256f2](https://github.com/FreeCAD/FreeCAD/pull/11294/commits/da256f2a94b76395195e19e59f74c5b6f67f36da)
It is to solve two problems. When you initialize a tool with a construction methode that is not default, then you have 2 problems : 
1 - The number of parameters / checkbox / combobox / OVP initialized is not the correct one. Which results in crash for Arc tool when you try to initiate 3 rim point arc.
2 - The combobox do not show the correct entry. Currently in master if you start the command 'centered rectangle' then the combobox shows 'diagonal'.

Comment (PaddleStroke) [reply src/Mod/Sketcher/Gui/DrawSketchDefaultWidgetController.h:256]: So if we keep the multiple tools in the toolbars (3 rim points arcs, centered rectangle, oblong) this is necessary.
But if we remove the tools, then its not necessary anymore.

**Comment by PaddleStroke** [comment]
I just seen there are multiple lint warning. I'll fix them

File: src/Mod/Sketcher/Gui/DrawSketchDefaultWidgetController.h (hunk 1 above)
Comment (abdullahtahiriyo) [reply src/Mod/Sketcher/Gui/DrawSketchDefaultWidgetController.h:256]: Ah! It is the overriding of the default construction mode on initialisation. Good then! Thanks.

**Comment by abdullahtahiriyo** [comment]
> * Cursor text (as this is linked to preferences). Also we did not decide what we want to do when OVP are enabled. My opinion is to disable the text when OVP are active. We don't need additional clutter.

I would implement it like you say. But I would do it transparently. Because you are going to need code at the tools for this, call a function there. The function will check the status of the parameter and show or not show depending on the parameter. If tomorrow we want all cursor texts in both modes, we just change that function. Does it sound ok to you?
